crates/common/consensus/src/electra/beacon_state.rs

```diff
@@ -1593,6 +1627,80 @@ impl BeaconState {
         Ok(())
     }
 
+    pub fn compute_sync_committee_rewards(
+        &self,
+        beacon_block: &SignedBeaconBlock,
+    ) -> anyhow::Result<HashMap<u64, u64>> {
```

Comment:
this function isn't used in `beacon_state.rs` is this code duplicate logic from where else?

Reply:
I have implemented this function in `handlers/block.rs`, then moved it to the beacon_state.rs in the previous refactor. Should i move it back?

Reply:
Most of the code in this function is duplicate logic from `process_sync_aggregate()` right.

Ideally `process_sync_aggregate()` and `compute_sync_committee_rewards()` use the same base function to do the raw calculations.

Reply:
Makes sense, will do the refactor for this.